### DISCUSSION TITLE
Establishing an Enterprise GPTCache accessed by Multiple Applications ?

### DISCUSSION THREAD

**Discussion by keeppace** [zilliztech/GPTCache#491] [Q&A]
As I read thru the GPTCache documentation - it seems application centric/focused.    

Architecturally - I am looking to put a caching architecture in front of an LLM - so that all applications can use the same Cache (rather than do it application by application) 

Just looking for ideas as to how to best implement such functionality

**Comment by SimFG** [comment]
@keeppace Is it like a cache server? 
When each application receives a request, it first queries the public cache service. If the cache hits, it directly responds to the request. Otherwise, it processes the request through LLM, returns the processing result to the cache, and responds to the user request.
